src/ml_gym/util/util.py

```diff
@@ -61,6 +106,154 @@ def create_system_info() -> Dict:
+    hyperparams: dict = None
+    loss_func: str = ""
+    optimizer: str = ""
+
+    def toJSON(self) -> Dict:
+        model_card ={}
+        model_card["hyperparams"] = self.hyperparams
+        model_card["loss_func"] = self.loss_func
+        model_card["optimizer"] = self.optimizer
+        return model_card
+
+@dataclass
+class EvalDetails:
+    loss_funcs: list = None
+    metrics: list = None
+
+    def toJSON(self) -> Dict:
+        model_card ={}
+        model_card["loss_funcs"] = self.loss_funcs
+        model_card["metrics"] = self.metrics
+        return model_card
+
+@dataclass
+class ModelCard:
+    model_details: ModelDetails
+    dataset_details: DatasetDetails
+    experiment_environment: ExperimentEnvironment
+    training_details: TrainingDetails
+    eval_details: EvalDetails
+
+    def toJSON(self) -> Dict:
+        model_card = {}
+        model_card["model_details"] = self.model_details.toJSON()
+        model_card["dataset_details"] = self.dataset_details.toJSON()
+        model_card["training_details"] = self.training_details.toJSON()
+        model_card["eval_details"] = self.eval_details.toJSON()
+        model_card["experiment_environment"] = self.experiment_environment.toJSON()
+        return model_card
+
+class SystemEnv:
+
+    @staticmethod
+    def create_model_card(grid_search_id: str, exp_config: dict, gs_config: dict, model = None) -> Dict:
+        """
+        Create Model card.
+        :params:
+                grid_search_id (str): Grid Search ID created for the run.
+                exp_config (dict): Experiment configuration.
+                gs_config (dict): Grid Search configuration.
+        :returns:
+                model_card (ModelCard): Model card object to be converted into json file.
+        """
+        
+        def update_model_details(grid_search_id: str, gs_config: dict, model) -> ModelDetails:
+            """
+            Function to initialize ModelDetails object.
+            :params:
+                    grid_search_id (str): Grid Search ID created for the run.
+                    gs_config (dict): Grid Search configuration.
+            :returns:
+                    obj (ModelDetails): initialized model details object.
+            """
+            try:
+                if model != None:
+                    pytorch_total_params = sum(p.numel() for p in model.parameters() if p.requires_grad)
+                else:
+                    pytorch_total_params = 0
+                train_date = datetime.now().strftime("%d-%m-%Y")
+                model_info = gs_config["model_info"]
+                return ModelDetails(model_description = model_info["model_description"], model_version = model_info["model_version"], grid_search_id = grid_search_id, train_date = train_date, source_repo = model_info["source_repo"], train_params= pytorch_total_params)
+            except Exception as e:
+                print("Error while fetching Model Details for Model card.", e)
```

Comment:
we should propagate the error and not "log" it as this low level in the call stack

Reply:
also a specific Exception would make more sense. What kind of things could go wrong here? 
As workaround we could catch for Exception and then reraise the Exception with a more specific exception type

Reply:
yeah exception handling is good. In this case, I think that if there is an error in the model card retrieval, it is fine to let the execution crash with an exception, cause in that case something is really wrong within the training run and it would make sense to kill it. What kind of cases can we expect, that would cause an error here?

Reply:
ok, I will then make the changes accordingly. The only case where we can get an error is if the YAML and JSON file structure changes and the places where I am trying to retrieve the information are broken or not there.